File: 0102.binary_tree_level_order_traversal/solution.py
from typing import List


from utils import TreeNode


class Solution:
    def level_order(self, root: TreeNode) -> List[List[int]]:
        """ 广度优先 """
        if not root:
            return []
        queue = [(root, 0)]
        ans = []
        while queue:
            node, level = queue.pop(0)
            if level > len(ans) - 1:
                ans.append([node.val])
            else:
                ans[level].append(node.val)
            if node.left:
                queue.append((node.left, level + 1))
            if node.right:
                queue.append((node.right, level + 1))
        return ans

File: 0102.binary_tree_level_order_traversal/solution.go
package leetcode0102

import "github.com/Ackerr/Algorithms/utils"

func levelOrder(root *utils.TreeNode) [][]int {
	ans := make([][]int, 0)
	if root == nil {
		return ans
	}
	queue := []*utils.TreeNode{root}
	for len(queue) > 0 {
		length := len(queue)
		temp := make([]int, length)
		for i := 0; i < length; i++ {
			set := queue[i]
			temp[i] = set.Val
			if set.Left != nil {
				queue = append(queue, set.Left)
			}
			if set.Right != nil {
				queue = append(queue, set.Right)
			}
		}
		ans = append(ans, temp)
		queue = queue[length:]
	}
	return ans
}
